kyaml/fn/framework/processors.go

```diff
@@ -382,29 +418,86 @@ func TemplatesFromDir(dirs ...pkger.Dir) TemplatesFunc {
 				if !strings.HasSuffix(info.Name(), ".template.yaml") {
 					return nil
 				}
-				name := path.Join(dir, info.Name())
-				f, err := pkger.Open(name)
+				t, err := parseTemplate(path.Join(dir, info.Name()))
 				if err != nil {
 					return err
 				}
-				defer f.Close()
 
-				b, err := ioutil.ReadAll(f)
+				pt = append(pt, t)
+				return nil
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+		return pt, nil
+	}
+}
+
+// SchemaDefinitionsFromFile returns a SchemaDefinitionFunc that will load schemas from the provided files.
+// This is a helper to facilitate providing custom resource schemas to a TemplateProcessor.
+func SchemaDefinitionsFromFile(files ...string) SchemaDefinitionFunc {
+	return func() ([]*spec.Definitions, error) {
+		var defs []*spec.Definitions
+		for _, filename := range files {
+			def, err := readSchemaJSON(filename)
+			if err != nil {
+				return nil, err
+			}
+			defs = append(defs, &def)
+		}
+		return defs, nil
+	}
+}
+
+// SchemaDefinitionsFromDir returns a SchemaDefinitionFunc that will load schemas from the provided directories.
+// This is a helper to facilitate providing custom resource schemas to a TemplateProcessor.
+func SchemaDefinitionsFromDir(dirs ...pkger.Dir) SchemaDefinitionFunc {
+	return func() ([]*spec.Definitions, error) {
+		var defs []*spec.Definitions
+		for i := range dirs {
+			dir := string(dirs[i])
+			err := pkger.Walk(dir, func(p string, info os.FileInfo, err error) error {
 				if err != nil {
 					return err
 				}
-				t, err := template.New(info.Name()).Parse(string(b))
+				if !strings.HasSuffix(info.Name(), ".json") {
+					return nil
+				}
+				def, err := readSchemaJSON(path.Join(dir, info.Name()))
 				if err != nil {
 					return err
 				}
-
-				pt = append(pt, t)
+				defs = append(defs, &def)
 				return nil
 			})
 			if err != nil {
 				return nil, err
 			}
 		}
-		return pt, nil
+		return defs, nil
+	}
+}
+
+func readSchemaJSON(filename string) (spec.Definitions, error) {
+	f, err := pkger.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var schema spec.Schema
+	err = schema.UnmarshalJSON(b)
+	if err != nil {
+		return nil, err
+	}
+
+	if schema.Definitions != nil {
+		return schema.Definitions, nil
 	}
+	return nil, errors.Errorf("schema did not contain any definitions")
```

Comment:
Since we're already providing both a file-based and a directory-based helper, I think it's reasonable to require the directory to be schema-specific. Plus these helpers plug into an interface, so it should be pretty easy to write your own helper that does something bespoke.